notebooks/fs.ipynb

```diff
@@ -0,0 +1,1340 @@
+   "source": [
+    "With these instructions, `copy()` creates a copy of each given file with the given name and in the given subdirectory. Copies are created in the directory indicated by `target_dir`. Paths can be provided either as a string or <a href=\"https://docs.python.org/3/library/pathlib.html#pathlib.Path\">Path</a> object. Any directories in the targeted paths for copying will be created if they don't already exist. `True` is returned upon a successful copy."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "a6aff6e3-815c-496e-81d7-d8756be9c232",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "True"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "fs.copy(\n",
+    "    config=\"fixtures/fs/copy-config.yaml\",\n",
+    "    target_dir=Path(\"tmp/copy-target\")\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5c05369f-3cf6-4576-8689-df98ed9d151d",
+   "metadata": {},
+   "source": [
+    "Examining the target directory, we can see that the copies of the files have been made with their specified names and in their specified directories."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "e67fdb49-beef-4006-9e36-1a22829f21fc",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\u001b[01;34mtmp/copy-target\u001b[0m\n",
+      "├── \u001b[01;34mdata\u001b[0m\n",
+      "│   ├── file2-copy.txt\n",
+      "│   └── file3-copy.csv\n",
+      "└── file1-copy.nml\n",
+      "\n",
+      "2 directories, 3 files\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%bash\n",
+    "tree tmp/copy-target"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "90e17445-3d87-4894-8211-8c737f7579d6",
+   "metadata": {},
+   "source": [
+    "### Failing to copy\n",
+    "\n",
+    "A configuration can be provided as a dictionary instead as this example demonstrates. However, `missing-file.nml` does not exist. The function provides a warning and returns `False`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "b1fa6662-c4f3-4f7a-9b5d-8ee258cd6e0e",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "File fixtures/fs/missing-file.nml: State: Pending (EXTERNAL)\n",
+      "Copy fixtures/fs/missing-file.nml -> tmp/copy-target/missing-copy.nml: Final state: Pending\n",
+      "File copies: Final state: Pending\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "False"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "fs.copy(\n",
+    "    config={\"missing-copy.nml\":\"fixtures/fs/missing-file.nml\"},\n",
+    "    target_dir=\"tmp/copy-target\"\n",
+    ")"
```

Comment:
This one is interesting: The output cell changes if you re-run the cell. I get three lines of output if I restart the Jupyter kernel and run the cell, but then only one line if I run the cell again. I think this might have to do with caching of task output and beyond the scope of this PR to address, but could you confirm that you see it too? I might know how to fix it if so, but in a different PR. I think it might confuse users to see different output each time they run a cell if they're not changing anything.

Reply:
I see the same change in output on my end when running this cell: three lines on the first run and then one line after.